.github/workflows/ruff.yml

```diff
@@ -11,15 +11,15 @@ on:
       - master
 
 jobs:
-  check-black-formatting:
+  check-ruff-formatting:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - uses: actions/setup-python@v5
       with:
         python-version: 3.8
         cache: 'pip'
-        cache-dependency-path: '.github/workflows/black.yml'
-    - run: pip install black==22.3.0
-    - run: black --diff .
-    - run: black --check .
+        cache-dependency-path: '.github/workflows/ruff.yml'
+    - run: pip install ruff==0.11.8
+    - run: ruff format --diff .
+    - run: ruff format --check .
```

Comment:
I think this is not needed, `--diff` already does this? Please also check the other repos, and keep it consistent.